pkg/apis/v1alpha2/aksnodeclass_status.go

```diff
@@ -68,3 +70,27 @@ func (in *AKSNodeClass) GetConditions() []status.Condition {
 func (in *AKSNodeClass) SetConditions(conditions []status.Condition) {
 	in.Status.Conditions = conditions
 }
+
+// Will return nil if the the KubernetesVersion is considered valid to use,
+// otherwise will return an error detailing the reason of failure.
+//
+// Ensures
+// - The AKSNodeClass is non-nil
+// - The AKSNodeClass' KubernetesVersionReady Condition is true
+// - The condition's ObservedGeneration is up to date with the latest spec generation
+// - The KubernetesVersion is initialized and non-empty
+func (in *AKSNodeClass) KubernetesVersionReadyAndValid() error {
+	if in == nil {
+		return fmt.Errorf("NodeClass is nil, meaning %s is consequently unready", ConditionTypeKubernetesVersionReady)
+	}
+	kubernetesVersionStatusCondition := in.StatusConditions().Get(ConditionTypeKubernetesVersionReady)
+	if !kubernetesVersionStatusCondition.IsTrue() {
+		return fmt.Errorf("NodeClass condition %s is not ready with status %s ", ConditionTypeKubernetesVersionReady, kubernetesVersionStatusCondition.GetStatus())
+		// TODO: this needs to be uncommented as soon as we update core to 1.1.x, but until then would make tests, and code checks fail.
```

Comment:
DO NOT RESOLVE THIS COMMENT UNTIL THE TODO IS ADDRESSES.

This comment needs to be addressed before completing this PR, as we want to handle ObservedGeneration. The code here is commented out for testing purposes while we wait for `1.1.x` core bump.

Reply:
Resolved, and uncommented with TODO removed, as core bump occurred here:
https://github.com/Azure/karpenter-provider-azure/pull/601